include/openenclave/internal/cpuid.h

```diff
@@ -21,16 +22,24 @@
 #define OE_CPUID_RDSEED_FEATURE 0x00040000u /* Leaf 7, subleaf 0, EBX */
 
 /**
- * The list of cpuid leafs that are emulated.
- * Currently 0, 1, 4, 7 leafs are emulated, consistent with Intel SDK.
- * Note: leaf 1 is used by mbedtls to determine aesni support.
- * The higher 8 bits of ebx for leaf 1 is the current processor id (initial APIC
+ * Checke if a cpuid leaf is emulated.
+ * Currently 0, 1, 4, 7, 0x80000000 and 0x80000001 leaves are
+ * emulated. Note: leaf 1 is used by mbedtls to determine aesni support. The
+ * higher 8 bits of ebx for leaf 1 is the current processor id (initial APIC
  * id). Since CPUID emulation returns cached values, this higher 8 bits of ebx
  * should not be relied upon for leaf 1.
  */
+extern const uint32_t supported_cpuid_leaves[OE_CPUID_LEAF_COUNT];
 OE_INLINE bool oe_is_emulated_cpuid_leaf(uint32_t leaf)
 {
-    return (leaf == 0) || (leaf == 1) || (leaf == 4) || (leaf == 7);
+    // OE_CPUID_TRACE_ENUM_LEAF is used only in the exception test as an
+    // unsupported leaf
+    if (leaf == OE_CPUID_TRACE_ENUM_LEAF)
```

Comment:
There should not need to be a special case to handle this. The point of testing an arbitrary leaf like 0x14 is to verify the negative test case, not that the implementation should implement special behavior for it. 

In fact, it would be better if `OE_CPUID_TRACE_ENUM_LEAF` is declared in the body of sigill_handling.c, which is the only place that it is used, instead of in the general header here.